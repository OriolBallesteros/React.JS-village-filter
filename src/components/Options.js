import React from 'react';
import '../styles/Options.scss';

const Options = props => {
    return (
        <select className='Option' onChange={props.onHandleChange}>
            {props.list &&
                props.list.map(item => <option key={item} value={item}>{item}</option>)}
        </select>
    );
};

export default Options;