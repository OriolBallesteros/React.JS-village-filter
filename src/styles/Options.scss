.Option {
    border-radius: 5px;
    margin: 10px;

    option{
        text-align: center;
    }
}