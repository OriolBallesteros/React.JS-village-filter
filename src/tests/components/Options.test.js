import React from 'react';
import { shallow } from 'enzyme';
import Options from '../../components/Options';

it('should render App component correctly', () => {
    const wrapper = shallow(<Options 
        onHandleChange={jest.fn()} 
        list={[1, 2, 3]} />);

    expect(wrapper).toMatchSnapshot();
});