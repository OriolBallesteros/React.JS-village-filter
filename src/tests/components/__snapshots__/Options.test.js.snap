// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render App component correctly 1`] = `
<select
  className="Option"
  onChange={[MockFunction]}
>
  <option
    key="1"
    value={1}
  >
    1
  </option>
  <option
    key="2"
    value={2}
  >
    2
  </option>
  <option
    key="3"
    value={3}
  >
    3
  </option>
</select>
`;
